pkg/util/indexparamcheck/scann_checker.go

```diff
@@ -0,0 +1,41 @@
+package indexparamcheck
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// scaNNChecker checks if a SCANN index can be built.
+type scaNNChecker struct {
+	ivfBaseChecker
+}
+
+// CheckTrain checks if SCANN index can be built with the specific index parameters.
+func (c *scaNNChecker) CheckTrain(params map[string]string) error {
```

Comment:
No need to use pointer receiver here.

Reply:
should we change ivf_pq_checker, ivf_sq_checker as well? The code is copied here...